iphone/Classes/TiViewProxy.m

```diff
@@ -167,10 +169,25 @@ - (BOOL)belongsToContext:(id<TiEvaluator>)context
 
 - (void)add:(id)arg
 {
+  TiUIWindowProxy *windowProxy = nil;
+  if ([self isKindOfClass:[TiUIWindowProxy class]]) {
+    //Added a transparent safeAreaViewProxy above window for safe area layouts if shouldExtendSafeArea is false. All views added on window will be added on safeAreaViewProxy. Layouts of safeAreaViewProxy is getting modified wherever required.
+     windowProxy = (TiUIWindowProxy *)self;
+     windowProxy.shouldExtendSafeArea = [TiUtils boolValue:[self valueForUndefinedKey:@"extendSafeArea"]
+                                             def:NO];
+    if (!windowProxy.safeAreaViewProxy && !windowProxy.shouldExtendSafeArea) {
+      windowProxy.safeAreaViewProxy = [[TiUIViewProxy alloc] _initWithPageContext:[self pageContext] args:nil];
```

Comment:
May need to be `[[TiUIViewProxy alloc] _initWithPageContext:[self pageContext]] autorelease]`